### ISSUE TITLE
union result is not same as mysql 

### ISSUE THREAD

**Issue by H-ZeX** [pingcap/tidb#11590]
## Bug Report

1. What did you do?
If possible, provide a recipe for reproducing the error.

```sql
create table t(a int);
create table s(a int unsigned);

insert into t value(-1);
insert into s value(1);
```

2. What did you expect to see?

```
select * from t union all select * from s;
+------+
| a    |
+------+
|   -1 |
|    1 |
+------+

select * from s union all select * from t;
+------+
| a    |
+------+
|   -1 |
|    1 |
+------+
```

3. What did you see instead?

```
select * from t union all select * from s;
+------+
| a    |
+------+
|   -1 |
|    1 |
+------+

select * from s union all select * from t;
+------+
| a    |
+------+
|    0 |
|    1 |
+------+
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

```
Release Version: v4.0.0-alpha-15-gd2b538718
Git Commit Hash: d2b5387183a591847f952aa4e3a3c4597c34dcba
Git Branch: HEAD
UTC Build Time: 2019-08-02 08:50:35
GoVersion: go version go1.12.6 linux/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false
```
```
mysqld  Ver 8.0.13 for linux-glibc2.12 on x86_64 (MySQL Community Server - GPL)
````

5. relate issue
https://github.com/pingcap/tidb/issues/7075

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

**Comment by SunRunAway** [comment]
Hi, @H-ZeX
could you provide the version of your MySQL server?

**Comment by H-ZeX** [comment]
> Hi, @H-ZeX
> could you provide the version of your MySQL server?
```
mysqld  Ver 8.0.13 for linux-glibc2.12 on x86_64 (MySQL Community Server - GPL)
```

**Comment by SunRunAway** [comment]
TiDB aims to compatible with MySQL5.7 so it seems to be fixed in #7075

**Comment by SunRunAway** [comment]
Hi, @H-ZeX 
I'm closing this issue now, feel free to reopen it if I'm wrong.
